crates/ruff_linter/src/rules/pyflakes/rules/unused_import.rs

```diff
@@ -138,9 +165,32 @@ fn is_first_party(qualified_name: &str, level: u32, checker: &Checker) -> bool {
     }
 }
 
+/// Find the `Expr` for top level `__all__` bindings.
+fn find_dunder_all_exprs<'a>(semantic: &'a SemanticModel) -> Vec<&'a ast::Expr> {
+    semantic
+        .global_scope()
+        .get_all("__all__")
+        .filter_map(|binding_id| {
+            let binding = semantic.binding(binding_id);
+            let stmt = match binding.kind {
+                BindingKind::Export(_) => binding.statement(semantic),
+                _ => None,
+            }?;
+            match stmt {
```

Comment:
```suggestion
            if !binding.kind.is_export() {
                return None;
            }
            match binding.statement(semantic)? {
```

Reply:
I think it's clearer to downcast from binding to statement in one statement here